components/script/dom/htmlinputelement.rs

```diff
@@ -875,6 +875,24 @@ impl HTMLInputElement {
                 content.strip_newlines();
                 content.strip_leading_and_trailing_ascii_whitespace();
             }
+            atom!("color") => {
+                let mut textinput = self.textinput.borrow_mut();
+                let content = textinput.get_content();
+                let is_valid = {
+                    let mut chars = content.chars();
```

Comment:
Why does this need to be mutable?

Reply:
The iterator needs to be mutable because we're calling `.next()` on it to discard the `#` before checking if every other character in it is a valid hex digit.